File: apps/api/aqapi/family/entity/exp_by_level_entity.py
from datetime import datetime
from sqlalchemy import Column, Integer, DateTime, ForeignKey, func
from sqlalchemy.orm import relationship, Mapped, mapped_column
from aqapi.core.entity.base_entity import BaseEntity
from aqapi.core.entity.base_translation_entity import BaseTranslationEntity
from aqapi.core.config.db_config import DB_CONF


class ExpByLevelEntity(BaseEntity):
    """経験値レベル設定エンティティ"""

    __tablename__ = "exp_by_level"

    family_id: Mapped[int] = mapped_column(Integer, ForeignKey("families.id", ondelete="CASCADE"), nullable=False, comment="家族ID")
    level: Mapped[int] = mapped_column(Integer, nullable=False, comment="レベル")
    exp: Mapped[int] = mapped_column(Integer, nullable=False, comment="レベルに必要な経験値")

    family = relationship("FamiliesEntity", foreign_keys=[family_id])

File: packages/backend/src/features/family/entity/expByLevelEntity.ts
import { BaseTransactionEntity } from "@backend/core/entity/baseTransactionEntity";
import {
  Entity,
  Column,
  ManyToOne,
  JoinColumn,
} from "typeorm";
import { FamilyEntity } from "./familyEntity";

@Entity("exp_by_level")
export class ExpByLevelEntity extends BaseTransactionEntity {
  @Column({ name: "family_id", type: "int", nullable: false, comment: "家族ID" })
  familyId!: number;
  @Column({ name: "level", type: "int", nullable: false, comment: "レベル" })
  level!: number;
  @Column({ name: "required_exp", type: "int", nullable: false, comment: "必要経験値" })
  requiredExp!: number;

  @ManyToOne(() => FamilyEntity, { onDelete: "CASCADE" })
  @JoinColumn({ name: "family_id", referencedColumnName: "id", foreignKeyConstraintName: "fk_exp_by_level_family_id" })
  family?: FamilyEntity;

}
